libcudacxx/include/cuda/std/detail/libcxx/include/__cccl/visibility.h

```diff
@@ -0,0 +1,42 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of libcu++, the C++ Standard Library for your entire system,
+// under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2023 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef __CCCL_VISIBILITY_H
+#define __CCCL_VISIBILITY_H
+
+#ifndef __CCCL_CONFIG
+#error "<__cccl/visibility.h> should only be included in from <cuda/__cccl_config>"
+#endif // __CCCL_CONFIG
+
+// We want to ensure that all warning emmiting from this header are supressed
+// We define cub and thrust kernels as hidden. However, this triggers errors about missing external linkage iff the
+// definition of the _CCCL_ATTRIBUTE_HIDDEN macro is not in a system header :shrug:
+#if defined(_CCCL_COMPILER_NVHPC)
+#pragma GCC system_header
+#else // ^^^ _CCCL_COMPILER_NVHPC ^^^ / vvv !_CCCL_COMPILER_NVHPC vvv
+_CCCL_FORCE_SYSTEM_HEADER
+#endif // !_CCCL_COMPILER_NVHPC
+
+// Enable us to hide kernels
+#if defined(_CCCL_COMPILER_MSVC)
+#  define _CCCL_ATTRIBUTE_HIDDEN
+#elif defined(_CCCL_COMPILER_NVRTC)
+#  define _CCCL_ATTRIBUTE_HIDDEN
+#elif defined(_CCCL_COMPILER_NVHPC)
+#  define _CCCL_ATTRIBUTE_HIDDEN
+#else // ^^^ _CCCL_COMPILER_NVRTC ^^^ / vvv _CCCL_COMPILER_NVRTC vvv
```

Comment:
important: I'm not confident about changing kernel attributes here. Could you give a try to something along the following lines instead?
```cpp
#pragma nv_diag_suppress 1407
```

Reply:
turns out the nvc++ diagnostic suppression macros are equally broken :(